fireworks/scripts/lpad_run.py

```diff
@@ -691,6 +692,115 @@ def maintain(args):
     lp.maintain(args.infinite, args.maintain_interval)
 
 
+def orphaned(args):
+    # get_fws
+    lp = get_lp(args)
+    if sum([bool(x) for x in [args.fw_id, args.name, args.state, args.query]]) > 1:
+        raise ValueError('Please specify exactly one of (fw_id, name, state, query)')
+    if sum([bool(x) for x in [args.fw_id, args.name, args.state, args.query]]) == 0:
+        args.query = '{}'
+        args.display_format = args.display_format if args.display_format else 'ids'
+    if sum([bool(x) for x in [args.fw_id, args.name, args.qid]]) > 1:
+        raise ValueError('Please specify exactly one of (fw_id, name, qid)')
+    else:
+        args.display_format = args.display_format if args.display_format else 'more'
+
+    if args.fw_id:
+        query = {'fw_id': {"$in": args.fw_id}}
+    elif args.name and not args.launches_mode:
+        query = {'name': args.name}
+    elif args.state:
+        query = {'state': args.state}
+    elif args.query:
+        query = ast.literal_eval(args.query)
+    else:
+        query = None
+
+    if args.sort:
+        sort = [(args.sort, ASCENDING)]
+    elif args.rsort:
+        sort = [(args.rsort, DESCENDING)]
+    else:
+        sort = None
+
+    if args.qid:
+        fw_ids = lp.get_fw_ids_from_reservation_id(args.qid)
+        if query:
+            query['fw_id'] = {"$in": fw_ids}
+            fw_ids = lp.get_fw_ids(query, sort, args.max, launches_mode=args.launches_mode)
+    else:
+        fw_ids = lp.get_fw_ids(query, sort, args.max,
+                               launches_mode=args.launches_mode)
+
+    # get_wfs
+    orphaned_fw_ids = []
+    for fw_id in fw_ids:
+        query = {'nodes': fw_id}
+        wf_ids = lp.get_wf_ids(query)
+        if len(wf_ids) == 0:
+            orphaned_fw_ids.append(fw_id)
+
+    fws = []
+    if args.display_format == 'ids':
+        fws = orphaned_fw_ids
+    elif args.display_format == 'count':
+        fws = [orphaned_fw_ids]
+    else:
+        for id in orphaned_fw_ids:
+            fw = lp.get_fw_by_id(id)
+            d = fw.to_dict()
+            d['state'] = d.get('state', 'WAITING')
+            if args.display_format == 'more' or args.display_format == 'less':
+                if 'archived_launches' in d:
+                    del d['archived_launches']
+                del d['spec']
+            if args.display_format == 'less':
+                if 'launches' in d:
+                    del d['launches']
+            fws.append(d)
+    if len(fws) == 1:
+        fws = orphaned_fw_ids[0]
+
+    if args.remove:  # snippet from lpad.delete_wf
```

Comment:
if there is more complex code that is duplicated in this, I think it is better to just have a helper function that both functions reference. Otherwise if we make a fix or change to one of the functions (e.g. lpad.delete_wf) the behavior won't update in this function.

Reply:
Suggested helper is lp.delete_fws in https://github.com/materialsproject/fireworks/pull/404/commits/72e88889fc09417dd33fcdd5fc719200e3ceac45.